### PR TITLE
[Cloud PC]Add external partner settings

### PR DESCRIPTION
**Pull request by Shaowei-Dong**

### CODE REVIEW

File: api-reference/beta/resources/cloudpcexternalpartnersetting.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+doc_type: resourcePageType
+---
+
+# cloudPcExternalPartnerSetting resource type
+
+Namespace: microsoft.graph
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents a external partner setting on Cloud PC.
```
Comment (Yitengyu) [inline api-reference/beta/resources/cloudpcexternalpartnersetting.md:16]: should be `an` external partner setting

File: api-reference/beta/api/virtualendpoint-post-externalpartnersettings.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+HTTP/1.1 201 Created
+Content-Type: application/json
+
+{
+  "@odata.type": "#microsoft.graph.cloudPcExternalPartnerSetting",
+  "id": "b3548526-e615-3785-3118-be70b3968ec5",
+  "partnerId": "198d7140-80bb-4843-8cc4-811377a49a92",
+  "enableConnection": true,
+  "lastSyncDateTime": "2020-11-03T12:43:14Z",
+  "status": "active",
```
Comment (Yitengyu) [inline api-reference/beta/api/virtualendpoint-post-externalpartnersettings.md:96]: It should be `available` once it's created

File: api-reference/beta/api/virtualendpoint-list-externalpartnersettings.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      "status": "active",
+      "statusDetails": "The external partner is active"
+    },
+    {
+      "@odata.type": "#microsoft.graph.cloudPcExternalPartnerSetting",
+      "id": "dc6422cb-3001-45a7-9dcd-21207eea6b0e",
+      "partnerId": "459a0e56-da26-4ba1-a729-8eeef733425b",
+      "enableConnection": true,
+      "lastSyncDateTime": "2020-11-03T12:43:14Z",
+      "status": "unresponsive",
```
Comment (Yitengyu) [inline api-reference/beta/api/virtualendpoint-list-externalpartnersettings.md:93]: `unresponsive` is not of the status enum

File: api-reference/beta/toc.yml [hunk 4]
```diff
@@ -10225,6 +10225,17 @@ items:
             href: api/cloudpcorganizationsettings-get.md
           - name: Update
             href: api/cloudpcorganizationsettings-update.md
+        - name: Cloud PC external partner settings
```
Comment (timefrozen19) [inline api-reference/beta/toc.yml:10228]: No s to be consistent

File: api-reference/beta/api/virtualendpoint-post-externalpartnersettings.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+|Content-Type|application/json. Required.|
+
+## Request body
+In the request body, supply a JSON representation of the [cloudPcExternalPartnerSetting](../resources/cloudpcexternalpartnersetting.md) object.
+
+You can specify the following properties when creating a **cloudPcExternalPartnerSetting**.
+
+|Property|Type|Description|
+|:---|:---|:---|
+|partnerId|String|The external partner ID. Required|
```
Comment (timefrozen19) [inline api-reference/beta/api/virtualendpoint-post-externalpartnersettings.md:49]: period at the end

File: api-reference/beta/api/virtualendpoint-post-externalpartnersettings.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+}
+-->
+``` http
+POST https://graph.microsoft.com/beta/deviceManagement/virtualEndpoint/externalPartnerSettings
+Content-Type: application/json
+
+{
+  "@odata.type": "#microsoft.graph.cloudPcExternalPartnerSetting",
+  "partnerId": "198d7140-80bb-4843-8cc4-811377a49a92",
+  "enableConnection": true,
```
Comment (timefrozen19) [inline api-reference/beta/api/virtualendpoint-post-externalpartnersettings.md:73]: extra comma

File: api-reference/beta/api/cloudpcexternalpartnersetting-get.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+  "name": "get_cloudpcexternalpartnersetting"
+}
+-->
+``` http
+GET https://graph.microsoft.com/beta/deviceManagement/virtualEndpoint/externalPartnerSettings/{id}
+```
+
+
+### Response
+>**Note:** The response object shown here might be shortened for readability.
```
Comment (RuiHou105) [inline api-reference/beta/api/cloudpcexternalpartnersetting-get.md:65]: You can remove this extra note.

File: api-reference/beta/api/cloudpcexternalpartnersetting-update.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+{
+  "@odata.type": "#microsoft.graph.cloudPcExternalPartnerSetting",
+  "enableConnection": true
+}
+```
+
+
+### Response
+>**Note:** The response object shown here might be shortened for readability.
```
Comment (RuiHou105) [inline api-reference/beta/api/cloudpcexternalpartnersetting-update.md:76]: You can remove this extra note here and other places within this PR.

File: api-reference/beta/api/virtualendpoint-post-externalpartnersettings.md [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+## Request body
+In the request body, supply a JSON representation of the [cloudPcExternalPartnerSetting](../resources/cloudpcexternalpartnersetting.md) object.
+
+You can specify the following properties when creating a **cloudPcExternalPartnerSetting**.
+
+|Property|Type|Description|
+|:---|:---|:---|
+|partnerId|String|The external partner ID. Required|
+|enableConnection|Boolean|Enable or disable external partner connection. If true, external partner api will accept incoming calls from external partner. Required.|
+
```
Comment (RuiHou105) [inline api-reference/beta/api/virtualendpoint-post-externalpartnersettings.md:51]: As property `status `is required, why not provide it when creating a cloudPcExternalPartnerSetting oject?

File: api-reference/beta/api/cloudpcexternalpartnersetting-get.md [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "@odata.type": "#microsoft.graph.cloudPcExternalPartnerSetting",
+  "id": "b3548526-e615-3785-3118-be70b3968ec5",
+  "partnerId": "198d7140-80bb-4843-8cc4-811377a49a92",
+  "enableConnection": true,
+  "lastSyncDateTime": "2020-11-03T12:43:14Z",
+  "status": "active",
```
Comment (RuiHou105) [inline api-reference/beta/api/cloudpcexternalpartnersetting-get.md:82]: The enum `cloudPcExternalPartnerStatus `only includes: notAvailable, available, healthy, unhealthy. But why you use active in some examples of this PR?

**Review by RuiHou105 — CHANGES_REQUESTED** [review]
@Shaowei-Dong  
1.could you please resolve the comments in your PR? 
2.Please follow this [guidance ](https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211772/Content-review-process-summary)to create a roadmap workitem and then attach it into the description here. Thanks.

File: api-reference/beta/api/virtualendpoint-post-externalpartnersettings.md (hunk 9 above)
Comment (Shaowei-Dong) [reply api-reference/beta/api/virtualendpoint-post-externalpartnersettings.md:51]: status is required in this entity, but the value comes from some mechanism so user can't assign any value to it.

File: api-reference/beta/api/cloudpcexternalpartnersetting-get.md (hunk 10 above)
Comment (Shaowei-Dong) [reply api-reference/beta/api/cloudpcexternalpartnersetting-get.md:82]: Updated

File: api-reference/beta/resources/cloudpcexternalpartnersetting.md (hunk 1 above)
Comment (Shaowei-Dong) [reply api-reference/beta/resources/cloudpcexternalpartnersetting.md]: updated

File: api-reference/beta/api/virtualendpoint-post-externalpartnersettings.md (hunk 2 above)
Comment (Shaowei-Dong) [reply api-reference/beta/api/virtualendpoint-post-externalpartnersettings.md]: updated

File: api-reference/beta/api/virtualendpoint-list-externalpartnersettings.md (hunk 3 above)
Comment (Shaowei-Dong) [reply api-reference/beta/api/virtualendpoint-list-externalpartnersettings.md]: updated

File: api-reference/beta/toc.yml (hunk 4 above)
Comment (Shaowei-Dong) [reply api-reference/beta/toc.yml]: removed

File: api-reference/beta/api/virtualendpoint-post-externalpartnersettings.md (hunk 5 above)
Comment (Shaowei-Dong) [reply api-reference/beta/api/virtualendpoint-post-externalpartnersettings.md]: added

File: api-reference/beta/api/virtualendpoint-post-externalpartnersettings.md (hunk 6 above)
Comment (Shaowei-Dong) [reply api-reference/beta/api/virtualendpoint-post-externalpartnersettings.md]: removed

File: api-reference/beta/api/cloudpcexternalpartnersetting-get.md (hunk 7 above)
Comment (Shaowei-Dong) [reply api-reference/beta/api/cloudpcexternalpartnersetting-get.md]: removed

**Comment by RuiHou105** [comment]
@Shaowei-Dong 
1.Some comments are not resolved. Please help resolve them to move forward the review process.
2.Please create a roadmap workitem for your change and attach the workitem NO. in the comment. Here is an example:

**Review by RuiHou105 — APPROVED** [review]

File: api-reference/beta/resources/virtualendpoint.md [hunk 11]
```diff
@@ -36,6 +36,8 @@ Use the Cloud PC API to provision and manage virtual desktops for employees in a
 |[List supportedRegions](../api/virtualendpoint-list-supportedregions.md)|[cloudPcSupportedRegion](../resources/cloudpcsupportedregion.md) collection|List properties and relationships of the [cloudPcSupportedRegion](../resources/cloudpcsupportedregion.md) objects.|
 |[List servicePlans](../api/virtualendpoint-list-serviceplans.md)|[cloudPcServicePlan](../resources/cloudpcserviceplan.md) collection|List properties and relationships of the [cloudPcServicePlan](../resources/cloudpcserviceplan.md) objects.|
 |[List snapshots](../api/virtualendpoint-list-snapshots.md)|[cloudPcSnapshot](../resources/cloudpcsnapshot.md) collection|Get a list of [cloudPcSnapshot](../resources/cloudpcsnapshot.md) objects and their properties.|
+|[List externalPartnerSettings](../api/virtualendpoint-list-externalpartnersettings.md)|[cloudPcExternalPartnerSetting](../resources/cloudpcexternalpartnersetting.md) collection|Get a list of the [cloudPcExternalPartnerSetting](../resources/cloudpcexternalpartnersetting.md) objects and their properties.|
```
Comment (AshleyYangSZ) [inline api-reference/beta/resources/virtualendpoint.md:39]: Will externalPartnerSettings support get by ID,  update and delete API? If yes please add here, if no, please remove the get, update, delete API files.

**Review by AshleyYangSZ — CHANGES_REQUESTED** [review]
Do you support get,update, and delete externalPartnerSetting APIs? If no please remove these files

File: api-reference/beta/resources/virtualendpoint.md (hunk 11 above)
Comment (Shaowei-Dong) [reply api-reference/beta/resources/virtualendpoint.md:39]: Do you mean here? I've added them in cloudpcexternalpartnersetting.md file.

**Review by AshleyYangSZ — APPROVED** [review]

**Comment by JarbasHorst** [comment]
Hello @Shaowei-Dong, I reviewed your PR and left some comments. Please, take a look when you will have time.

Also, to avoid merge conflicts, please merge PR [17664](https://github.com/microsoftgraph/microsoft-graph-docs/pull/17664) into your branch to include my fixes and before addressing my comments.

File: api-reference/beta/api/cloudpcexternalpartnersetting-get.md [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+## Examples
+
+### Request
+<!-- {
+  "blockType": "request",
+  "name": "get_cloudpcexternalpartnersetting"
+}
+-->
+``` http
+GET https://graph.microsoft.com/beta/deviceManagement/virtualEndpoint/externalPartnerSettings/{id}
```
[the remark this answers is not in the document]
Comment (Shaowei-Dong) [reply api-reference/beta/api/cloudpcexternalpartnersetting-get.md]: thanks, updated.

File: api-reference/beta/api/cloudpcexternalpartnersetting-update.md [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+|Authorization|Bearer {token}. Required.|
+|Content-Type|application/json. Required.|
+
+## Request body
+[!INCLUDE [table-intro](../../includes/update-property-table-intro.md)]
+
+
+|Property|Type|Description|
+|:---|:---|:---|
+|enableConnection|Boolean|Enable or disable external partner connection. If true, external partner api will accept incoming calls from external partner. Required.|
```
[the remark this answers is not in the document]
Comment (Shaowei-Dong) [reply api-reference/beta/api/cloudpcexternalpartnersetting-update.md]: prefer to the first one, updated in 3 files.

**Comment by Shaowei-Dong** [comment]
@Lauragra @MSFT-Jarbas Please let me know if you any question/comments on this PR

**Comment by JarbasHorst** [comment]
Hi @isvargasmsft, can you please final check this PR and set the label "content review complete" if possible?

File: api-reference/beta/resources/cloudpcexternalpartnersetting.md [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+
+## Properties
+|Property|Type|Description|
+|:---|:---|:---|
+|enableConnection|Boolean|Enable or disable the connection to an external partner. If `true`, an external partner API will accept incoming calls from external partners. Required.|
+|id|String|The unique identifier for the Cloud PC external partner setting. Read-only.|
+|lastSyncDateTime|DateTimeOffset|Last data sync time for this external partner. The Timestamp type represents the date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 looks like this: '2014-01-01T00:00:00Z'.|
+|partnerId|String|The external partner ID.|
+|status|[cloudPcExternalPartnerStatus](#cloudpcexternalpartnerstatus-values)|The status of the connection to the external partner. The possible values are: `notAvailable`, `available`, `healthy`, `unhealthy`, `unknownFutureValue`.|
+|statusDetails|String|Status details message.|
```
Comment (isvargasmsft) [inline api-reference/beta/resources/cloudpcexternalpartnersetting.md:34]: Please indicate in the description of the properties which ones support $filter.

File: api-reference/beta/api/virtualendpoint-list-externalpartnersettings.md [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+|Authorization|Bearer {token}. Required.|
+
+## Request body
+Do not supply a request body for this method.
+
+## Response
+
+If successful, this method returns a `200 OK` response code and a collection of [cloudPcExternalPartnerSetting](../resources/cloudpcexternalpartnersetting.md) objects in the response body.
+
+## Examples
```
Comment (isvargasmsft) [inline api-reference/beta/api/virtualendpoint-list-externalpartnersettings.md:51]: Could you please include a second example using the $select query parameter? In general, we have been promoting the use of $select in Graph and that includes having examples in our docs.

**Review by isvargasmsft — CHANGES_REQUESTED** [review]
HEllo @Shaowei-Dong 

Excellent work in this PR. Before we merge, would you be so kind to include an example using the $select parameter to the list operation? 
In addition, please indicate which properties support filter in the description of the properties table. 

Kind regards, 

Isaac

File: api-reference/beta/resources/cloudpcexternalpartnersetting.md (hunk 14 above)
Comment (Shaowei-Dong) [reply api-reference/beta/resources/cloudpcexternalpartnersetting.md]: $filter added

File: api-reference/beta/api/virtualendpoint-list-externalpartnersettings.md (hunk 15 above)
Comment (Shaowei-Dong) [reply api-reference/beta/api/virtualendpoint-list-externalpartnersettings.md:51]: a second example added

**Comment by Shaowei-Dong** [comment]
@isvargasmsft please help review this PR, I expect this PR can be merged before the end of this month.

**Comment by Shaowei-Dong** [comment]
kindly ping @isvargasmsft  @JarbasHorst  @Lauragra

**Review by isvargasmsft — APPROVED** [review]

**Pull request merged**